File: sample-objc/ARGEARSample/UI/UIComponent/Preview/PreviewBottomFunctionView.m
//
//  PreviewBottomFunctionView.m
//  ARGEARSample
//
//  Created by Jihye on 10/09/2019.
//  Copyright © 2019 Seerslab. All rights reserved.
//

#import "PreviewBottomFunctionView.h"

@implementation PreviewBottomFunctionView

#define kPreviewBottomFunctionBackButtonName @"icBack"
#define kPreviewBottomFunctionSaveButtonName @"btnDownload"
#define kPreviewBottomFunctionCheckButtonName @"btnCheck"
#define kPreviewBottomFunctionShareButtonName @"icShare"

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

- (void)setImages:(ARGMediaRatio)ratio {
    NSMutableString *backButtonImageString = [NSMutableString stringWithString:kPreviewBottomFunctionBackButtonName];
    NSMutableString *saveButtonImageString = [NSMutableString stringWithString:kPreviewBottomFunctionSaveButtonName];
    NSMutableString *checkButtonImageString = [NSMutableString stringWithString:kPreviewBottomFunctionCheckButtonName];
    NSMutableString *shareButtonImageString = [NSMutableString stringWithString:kPreviewBottomFunctionShareButtonName];
    
    if (ratio == ARGMediaRatio_16x9) {
        // white
        [backButtonImageString appendString:@"White"];
        [saveButtonImageString appendString:@"White"];
        [checkButtonImageString appendString:@"White"];
        [shareButtonImageString appendString:@"White"];
    } else {
        // pink
        [backButtonImageString appendString:@"Black"];
        [saveButtonImageString appendString:@"Pink"];
        [checkButtonImageString appendString:@"Pink"];
        [shareButtonImageString appendString:@"Black"];
    }
    
    [_backButton setImage:[UIImage imageNamed:[NSString stringWithString:backButtonImageString]] forState:UIControlStateNormal];
    [_saveButton setImage:[UIImage imageNamed:[NSString stringWithString:saveButtonImageString]] forState:UIControlStateNormal];
    [_checkButton setImage:[UIImage imageNamed:[NSString stringWithString:checkButtonImageString]] forState:UIControlStateNormal];
    [_shareButton setImage:[UIImage imageNamed:[NSString stringWithString:shareButtonImageString]] forState:UIControlStateNormal];
}

- (void)showCheckButton {
    [_saveButton setHidden:YES];
    [_checkButton setHidden:NO];
}

- (IBAction)backButtonAction:(UIButton *)sender {
    if (_delegate && [_delegate respondsToSelector:@selector(backButtonAction)]) {
        [_delegate backButtonAction];
    }
}

- (IBAction)saveButtonAction:(UIButton *)sender {
    if (_delegate && [_delegate respondsToSelector:@selector(saveButtonAction)]) {
        [_delegate saveButtonAction];
    }
}

- (IBAction)checkButtonAction:(UIButton *)sender {
    if (_delegate && [_delegate respondsToSelector:@selector(checkButtonAction)]) {
        [_delegate checkButtonAction];
    }
}

- (IBAction)shareButtonAction:(UIButton *)sender {
    if (_delegate && [_delegate respondsToSelector:@selector(shareButtonAction)]) {
        [_delegate shareButtonAction];
    }
}

@end

File: sample-swift/ARGearSample/UI/UIComponent/Preview/PreviewBottomFunctionView.swift
//
//  PreviewBottomFunctionView.swift
//  ARGearSample
//
//  Created by Jihye on 2020/02/06.
//  Copyright © 2020 Seerslab. All rights reserved.
//

import UIKit
import ARGear

protocol PreviewBottomFunctionDelegate: class {
    func backButtonAction()
    func saveButtonAction()
    func checkButtonAction()
    func shareButtonAction()
}

class PreviewBottomFunctionView: UIView {
    
    let kPreviewBottomFunctionBackButtonName = "icBack"
    let kPreviewBottomFunctionSaveButtonName = "btnDownload"
    let kPreviewBottomFunctionCheckButtonName = "btnCheck"
    let kPreviewBottomFunctionShareButtonName = "icShare"
    
    weak var delegate: PreviewBottomFunctionDelegate?
    
    @IBOutlet weak var backButton: UIButton!
    @IBOutlet weak var saveButton: UIButton!
    @IBOutlet weak var checkButton: UIButton!
    @IBOutlet weak var shareButton: UIButton!
    
    public func setRatio(_ ratio: ARGMediaRatio) {
        var backButtonImageName = kPreviewBottomFunctionBackButtonName
        var saveButtonImageName = kPreviewBottomFunctionSaveButtonName
        var checkButtonImageName = kPreviewBottomFunctionCheckButtonName
        var shareButtonImageName = kPreviewBottomFunctionShareButtonName
        
        var imageSaveCheckColorString = ""
        var imageBackColorString = ""
        if ratio == ._16x9 {
            imageSaveCheckColorString = "White"
            imageBackColorString = "White"
        } else {
            imageSaveCheckColorString = "Pink"
            imageBackColorString = "Black"
        }
        backButtonImageName.append(imageBackColorString)
        saveButtonImageName.append(imageSaveCheckColorString)
        checkButtonImageName.append(imageSaveCheckColorString)
        shareButtonImageName.append(imageBackColorString)
        
        self.backButton.setImage(UIImage(named: backButtonImageName), for: .normal)
        self.saveButton.setImage(UIImage(named: saveButtonImageName), for: .normal)
        self.checkButton.setImage(UIImage(named: checkButtonImageName), for: .normal)
        self.shareButton.setImage(UIImage(named: shareButtonImageName), for: .normal)
    }
    
    @IBAction func backButtonAction(_ sender: UIButton) {
        guard let delegate = self.delegate else { return }
        
        delegate.backButtonAction()
    }
    
    @IBAction func saveButtonAction(_ sender: UIButton) {
        guard let delegate = self.delegate else { return }
        
        delegate.saveButtonAction()
    }
    
    @IBAction func checkButtonAction(_ sender: UIButton) {
        guard let delegate = self.delegate else { return }
        
        delegate.checkButtonAction()
    }
    
    @IBAction func shareButtonAction(_ sender: UIButton) {
        guard let delegate = self.delegate else { return }
        
        delegate.shareButtonAction()
    }
    
    func showCheckButton() {
        self.saveButton.isHidden = true
        self.checkButton.isHidden = false
    }
}
